File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_reachable: int = 0
        n: int = len(nums)
        
        for i, jump in enumerate(nums):
            if i > max_reachable:
                return False
            max_reachable = max(max_reachable, i + jump)
            if max_reachable >= n - 1:
                return True  
        
        return False  

        
            
        

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        if (n == 1){return true;}

        int jumpPower = nums[0];
        int pointer = 0;

        while (jumpPower > 0 && pointer < n-1) {
            jumpPower--;
            pointer++;
            if (nums[pointer] > jumpPower){
                jumpPower = nums[pointer];
            }
        }

        return pointer == n - 1;
    }
}